### PR TITLE
[ARM CPU] Fix fp16 RNN, CTCGreedyDecoder layers support

### PR DESCRIPTION
**Pull request by allnes**
### Details:
 - Fix fp16 instance for RNN layer (with help fp32 instance)

### Tickets:
 - CVS-120950

### CODE REVIEW

**Comment by wkobielx** [comment]
Tested manually: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/build-linux-macos_arm64/5427/
we are having 97 failing tests, seems related to 
https://jira.devtools.intel.com/browse/CVS-122699
https://github.com/openvinotoolkit/openvino/pull/20318

File: src/plugins/intel_cpu/src/nodes/rnn.cpp [hunk 1]
```diff
@@ -505,6 +506,9 @@ void RNN::configurePortDataTypes() {
     if (one_of(memory::data_type::bf16, inDataTypes[xIdx], inDataTypes[hIdx]))
         inDataTypes[xIdx] = outDataTypes[yIdx] = outDataTypes[hoIdx] = inDataTypes[hIdx] = memory::data_type::bf16; // required by oneDNN.
 
+    if (one_of(memory::data_type::f16, inDataTypes[xIdx], inDataTypes[hIdx]))
+        inDataTypes[xIdx] = outDataTypes[yIdx] = outDataTypes[hoIdx] = inDataTypes[hIdx] = memory::data_type::f32;
```
Comment (alvoron) [inline src/plugins/intel_cpu/src/nodes/rnn.cpp:510]: Is it oneDNN requirement?

Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/rnn.cpp:510]: @alvoron it's no oneDNN requirement, onednn can't work with f16 and we move to f32 calculation

Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/rnn.cpp]: @EgorDuplensky fixed as temporary solution

Comment (vpirogov) [reply src/plugins/intel_cpu/src/nodes/rnn.cpp]: oneDNN [does not have f16 support|https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html#implementation-limitations] on Intel CPUs due to lack of fast arithmetic. Supported data types are f32, bf16, and int8.

**Pull request closed without merging**